### ISSUE TITLE
undo management: separate blocky and browser 

### ISSUE THREAD

**Issue by jlaurens** [google/blockly#1915]
There is a concurrency between Blockly undo management and browser undo management.

Take the fixed blockly demo at `blockly/demos/fixed/index.html` and add a `textarea`. If you perform changes in both the text area and the workspace, undo/redo with key shortcut merges Blockly undo management and browser undo management.

Here are the steps:
0) load the modified page
1) enter some text in the text area
2) drag some block in the workspace
3) trigger an undo event with the key shortcut

Problem: both the text area and the workspace are reverted during the same operation.
Expected behaviour: when the workspace has the focus, the undo/redo actions should apply to the workspace only.

Ideally, when the workspace has the focus, the menu item undo/redo action of the browser should apply to the workspace only.

**Comment by samelhusseini** [comment]
Agreed, we should only handle keyboard events for Blockly when the user is focused on a Blockly component.

**Comment by gonfunko** [comment]
Related (?), Command Z in Safari will also reopen a closed tab, which happens in parallel with undoing the action in Blockly.

**Comment by gonfunko** [comment]
Fixed by #7946.
